php/commands/auth.php

```diff
@@ -25,10 +25,30 @@ class Auth_Command extends Terminus_Command {
    *
    * [--password=<value>]
    * : Log in non-interactively with this password. Useful for automation.
+   * [--session=<value>]
+   * : Authenticate using an existing session token
    * [--debug]
    * : dump call information when logging in.
    */
   public function login( $args, $assoc_args ) {
+    # First try to login using a session token if provided
+    if (isset($assoc_args['session'])) {
```

Comment:
So all this code looks valid, but from an organization this Class already has way to much logic in it and going forward I think we should defer as much logic as possible to the Terminus libraries. so for instance here It would be much more testable to do something like 

```
$session = Input::optional($assoc_args, 'session', false);
if ($session)  {
 Auth::loginFromSession($session);
} else {
  Auth::login($email, $password);
}
Terminus::line( "Logged in as ". $data['email'] );
Terminus::launch_self("art", array("fist"));
```